### ISSUE TITLE
Add rand(X) for Intervals and IntervalBoxs

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalArithmetic.jl#259]
```
Base.rand(X::Interval) = X.lo + rand() * diam(X)
Base.rand(X::IntervalBox) = rand.(X)

julia> X = IntervalBox(3..4, 5..6)
[3, 4] × [5, 6]

julia> rand(X)
2-element StaticArrays.SArray{Tuple{2},Float64,1,2}:
 3.0978371040211217
 5.05610679819454
```

**Comment by lbenet** [comment]
Nice idea!

> `Base.rand(X::IntervalBox) = rand.(X)`

What about having this returning an `IntervalBox`? The change is minimal:

```
Base.rand(X::IntervalBox) = IntervalBox( rand.(X) )
```

**Comment by lbenet** [comment]
Implemented in #260
